pkg/virtctl/vnc/vnc.go

```diff
@@ -84,37 +84,51 @@ type VNC struct {
 	clientConfig clientcmd.ClientConfig
 }
 
-func (o *VNC) Run(cmd *cobra.Command, args []string) error {
-	namespace, _, err := o.clientConfig.Namespace()
+func (o *VNC) setupLocalProxy(vmiName string) (*net.TCPListener, error) {
+	lnAddr, err := net.ResolveTCPAddr("tcp", fmt.Sprintf("127.0.0.1:%d", customPort))
 	if err != nil {
-		return err
+		return nil, fmt.Errorf("Can't resolve the address: %s", err.Error())
+	}
+	// The local tcp server is used to proxy the podExec websock connection to vnc client
+	ln, err := net.ListenTCP("tcp", lnAddr)
+	if err != nil {
+		return nil, fmt.Errorf("Can't listen on unix socket: %s", err.Error())
 	}
 
-	vmi := args[0]
+	return ln, nil
```

Comment:
I wonder if it isn't simpler to just flat out `return net.ListenTCP("tcp", lnAddr)` . 

Do you think the error string you're adding is important enough ?

Reply:
@maiqueb I would prefer to preserve the error message, I see that TCP listener errors are not so self explanatory so I think this extra info may help

Reply:
Fair enough.